### ISSUE TITLE
"rbw login: failed to log in to bitwarden instance: Username or password is incorrect. Try again."

### ISSUE THREAD

**Issue by WxNzEMof** [doy/rbw#145]
Today I noticed that I can no longer log in with rbw. It does not accept my password.

I noticed that others have reported this problem in #32 and #88, though originally those issues were filed for different problems, so I'm creating an issue for this specific problem. Hope this helps.

**Comment by rjp** [comment]
Also getting this.  Been digging in to see if I can figure out if things are going wrong in `rbw` but nothing useful yet.

**Comment by carlosala** [comment]
Same here!

**Comment by augustebaum** [comment]
Same also; this seemed to coincide with my swapping i3 for GNOME, but this issue tells me there is something deeper happening.

**Comment by JeffDess** [comment]
@augustebaum I don't think it's related to a DE/WM, I'm having the same in Plasma connecting to the default API. Maybe BW changed something recently?

**Comment by carlosala** [comment]
Was someone able to debug it? I really need rbw in my daily workflow🥲

**Comment by rjc** [comment]
Same here, with the exception that I _seem_ to be able to _register_ with `rbw register` but am being re-prompted for password constantly with `pinentry` claiming it isn't correct.

**Comment by rjp** [comment]
I've done some debugging and I think that `rbw` might have some issues (the `400` response, perhaps), I can't get anything other than a `500` back from the bitwarden API by hand following the API descriptions I've found elsewhere.  Will have a look at `vaultwarden` and see if that gives me any clues (assuming that vw is still compatible with the bw API!)

**Comment by rjp** [comment]
Have tested against `vaultwarden` and `rbw` works fine there.  Unhelpful, I know, but it really does seem like `bitwarden` are doing something weird that isn't specified in their API documentation (which is pretty terrible.)

**Comment by Msouza91** [comment]
Has anyone else tested #151 fix? I removed previous version of rbw and installed his revision with the proposed fix, but still getting the error message.

**Comment by rjc** [comment]
I've just tested the #151 fix - it **does** resolve the issue!

**Comment by joelsleeba** [comment]
#151 fixes it! yayy!!

**Comment by 00sapo** [comment]
Confirmed, it works

**Comment by Msouza91** [comment]
I might be doing something wrong here, I did and rbw purge, deleted the config file, uninstalled the previous version of rbw, then installed the one from the PR, killed rbw-agent and rebooted my computer, not all at the same time, but that is all what I have done so far and still getting the error message, I checked the cargo checkouts folder and I have the code with the fix in the version installed on my PC.
Thoroughly confused here :( I was so looking forward to try this with rofi-rbw, but seem to be suffering from a major case of Skill Issue here, not sure where else to look to solve this for me...

**Comment by rjc** [comment]
@Msouza91 Did your run `rbw register`?

**Comment by Msouza91** [comment]
@rjc Register returns  400 as well, after input of the id and secret.
![image](https://github.com/doy/rbw/assets/19522556/065e74c0-9636-4bfc-a82f-85fd3831caa9)

**Comment by rjc** [comment]
@Msouza91 Are you a _.eu_ user by any chance? If so, check the issue #148. If not, try on another machine where you previously hadn't used `rbw`. If it still doesn't work, try rotating the key and/or bumping _PBKDF2_ rounds. Other than that, I'm running out of ideas.

BTW, I've tested on _macOS 14.2.1_ and _OpenBSD -current_.

**Comment by Msouza91** [comment]
Regular bitwarden.com user, will try the PBK parameter and rotating keys, thanks for the pointers.

**Comment by 00sapo** [comment]
> I might be doing something wrong here, I did and rbw purge, deleted the config file, uninstalled the previous version of rbw, then installed the one from the PR, killed rbw-agent and rebooted my computer, not all at the same time, but that is all what I have done so far and still getting the error message, I checked the cargo checkouts folder and I have the code with the fix in the version installed on my PC. Thoroughly confused here :( I was so looking forward to try this with rofi-rbw, but I seem to be suffering from a major case of Skill Issue, not sure where else to look to solve this for me...

Try `which rbw` to check you have not other installations hiding the cargo command. Then, remove the repo in the .cargo folder and retry (I used --branch option, but --ref should work the same).

**Comment by Msouza91** [comment]
@00sapo Well, I only had cargo installed through cargo from the beginning, uninstalled the main version and used the -rev command from the PR to install it again, but didn't work for me, didn't get the time to test on a VM, but it wouldn't change much if it worked on the VM since I can't figure out what is wrong in my main machine.
![image](https://github.com/doy/rbw/assets/19522556/aef00afd-ba97-434c-9201-85ad353b7a78)

**Comment by 00sapo** [comment]
It says "rbw 1.8.3 is already installed". Remove it and install or use "--force"...

**Comment by tiagorangel2011** [comment]
I'm having this issue with rbw 1.9.0

**Comment by doy** [comment]
@tiagorangel2011: there have been a bunch of changes on the bitwarden server side similar to this since then, and 1.9.0 is a pretty old release. let me know if you still have issues with the latest version.

**Comment by ratnanil** [comment]
Fix #151 does not seem to work for me. I'm running `1.9.0`, which should include #151.. I don't know how to test a newer version, since none are available on https://git.tozt.net/rbw/releases/deb/, and I have no clue how to install `rbw` from GitHub. Any pointers?

``` 
(base) ➜  ~ rbw --version
rbw 1.9.0
(base) ➜  ~ rbw login    
rbw login: failed to log in to bitwarden instance: failed to parse JSON: EOF while parsing a value at line 1 column 0

```
